src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1188,15 +1189,32 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       if (capturedErrors === null) {
         capturedErrors = new Map();
       }
-      capturedErrors.set(boundary, {
-        componentName,
-        componentStack,
-        error,
-        errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
-        errorBoundaryFound,
-        errorBoundaryName,
-        willRetry,
-      });
+
+      if (__DEV__) {
+        capturedErrors.set(boundary, {
+          componentName,
+          componentStack,
+          error,
+          errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
+          errorBoundaryFound,
+          errorBoundaryName,
+          willRetry,
+          shouldIgnoreErrorMessage: error != null &&
+            typeof error.__reactShouldIgnoreErrorMessage === 'boolean' &&
+            error.__reactShouldIgnoreErrorMessage,
+        });
+      } else {
+        capturedErrors.set(boundary, {
+          componentName,
+          componentStack,
+          error,
+          errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
+          errorBoundaryFound,
+          errorBoundaryName,
+          willRetry,
+          shouldIgnoreErrorMessage: false,
+        });
+      }
```

Comment:
What I want to DCE is the extra checks

```js
error != null && error.__reactShouldIgnoreErrorMessage === true
```

that only happen in DEV.

But this convo moot anyway, because I'm going to remove this branch and just always ignore the error message in DEV.